index.js

```diff
@@ -97,78 +105,95 @@ class ServerlessAppsyncPlugin {
       scalar AWSIPAddress
     `;
 
-    return `${schema} ${awsTypes}`;
+    return config.map(apiConfig => `${apiConfig.schema} ${awsTypes}`);
   }
 
-  validateSchema() {
-    const schema = this.getSchema();
-    const ast = buildASTSchema(parse(schema));
-    const errors = validateSchema(ast);
+  validateSchemas() {
+    const schemas = this.getSchemas();
+    const asts = schemas.map(schema => buildASTSchema(parse(schema)));
+    const errors = asts.reduce((accumulatedErrors, currentAst) => {
+      const currentErrors = validateSchema(currentAst);
+      if (!currentErrors.length) {
+        return accumulatedErrors;
+      } else if (!accumulatedErrors.length) {
+        return currentErrors;
+      } else {
+        return accumulatedErrors.concat(currentErrors || []);
+      }
+    });
```

Comment:
Can you provide an initialValue to the `reduce` method?

According to the [documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce) :

```
If no initial value is supplied, the first element in the array will be used.
```

The following case 

```
else if (!accumulatedErrors.length)
```

will never occur, and `accumulatedErrors` will include `asts[0]`

Reply:
Good point. I set the initial value to an empty array and removed the accumulatedErrors base case and updated get-config to reflect the changes. If an empty configuration (i.e. plugin is enabled but nothing in config) exists, get-config will return an empty array which can be used to call `.map` on.